packages/ui/src/memberships/modals/EmailSubscriptionModal/EmailSubscriptionModal.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+
+import { useApi } from '@/api/hooks/useApi'
+import { TextMedium } from '@/common/components/typography'
+import { useMachine } from '@/common/hooks/useMachine'
+import { useModal } from '@/common/hooks/useModal'
+import { SignTransactionModal } from '@/common/modals/SignTransactionModal/SignTransactionModal'
+import { EmailSubscriptionModalCall } from '@/memberships/modals/EmailSubscriptionModal/index'
+
+import { EmailSubscriptionFormModal } from './EmaiSubscriptionFormModal'
+import { EmailSubscriptionMachine } from './machine'
+import { EmailSubscriptionForm } from './types'
+import { createBatch } from './utils'
+
+export const EmailSubscriptionModal = () => {
+  const { api } = useApi()
+  const {
+    hideModal,
+    modalData: { member },
+  } = useModal<EmailSubscriptionModalCall>()
+  const [state, send] = useMachine(EmailSubscriptionMachine)
+
```

Comment:
I think the sign modal can be triggered by simply calling:
```ts
const signature = await api.sign(member.controllerAccount, `${member.id}:${timestamp}`)
```

Keep in mind the `signup` mutation is:
```gql
signup(memberId: String! signature: String! timestamp: BigInt! name: String! email: String): String
```
(As documented in the [backend readme](https://github.com/Joystream/pioneer/tree/backend/packages/server#authentication)). So the signature and timestamp should be stored.